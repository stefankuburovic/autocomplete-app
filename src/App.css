.App {
  text-align: center;
  padding: 20px;
}

.autocomplete {
  position: relative;
  display: inline-block;
}

input {
  width: 300px;
  padding: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
  font-size: 16px;
}

.autocomplete-list {
  position: absolute;
  width: 100%;
  max-height: 150px;
  overflow-y: auto;
  border: 1px solid #ccc;
  border-top: none;
  list-style: none;
  padding: 0;
  margin: 0;
  display: none;
  background-color: white;
}
.autocomplete-list.active {
  display: block;
}

.autocomplete-list li.autocomplete-list_not-selectable {
  user-select: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -o-user-select: none;
  cursor: unset;
}

.autocomplete-list li {
  padding: 10px;
  cursor: pointer;
  transition: background-color 0.2s;
}

.autocomplete-list li:hover {
  background-color: #f0f0f0;
}