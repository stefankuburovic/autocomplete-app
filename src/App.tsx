import React from 'react';
import './App.css';
import AutoComplete from './components/AutoComplete';
import data from './data/dummy.json';

const App: React.FC = () => {
  const handleSelect = (selectedItem: string) => {
    // Handle the selected item (e.g., set it in state or perform some action)
    console.log(`Selected: ${selectedItem}`);
  };

  return (
      <div className="App">
        <h1>AutoComplete Example</h1>
        <AutoComplete data={data} onSelect={handleSelect} />
      </div>
  );
};

export default App;