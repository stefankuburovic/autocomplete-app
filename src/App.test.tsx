import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import App from './App';

test('renders the app header', () => {
  render(<App />);
  const headerElement = screen.getByText(/AutoComplete Example/i);
  expect(headerElement).toBeInTheDocument();
});

test('renders the input field', () => {
  render(<App />);
  const inputElement = screen.getByPlaceholderText('Type to search...');
  expect(inputElement).toBeInTheDocument();
});

test('starts with an empty input field', () => {
  render(<App />);
  const inputElement = screen.getByPlaceholderText('Type to search...');
  expect(inputElement).toHaveValue('');
});

test('displays suggestions when typing in the input field', () => {
  render(<App />);
  const inputElement = screen.getByPlaceholderText('Type to search...');
  fireEvent.change(inputElement, { target: { value: 'a' } });
  setTimeout(
  () => {
    const suggestionElement = screen.getByText(/Apple/i);
    expect(suggestionElement).toBeInTheDocument();
  }, 500);
});

test('selects an item when clicking on a suggestion', () => {
  render(<App />);
  const inputElement = screen.getByPlaceholderText('Type to search...');
  fireEvent.change(inputElement, { target: { value: 'B' } });
  setTimeout(() => {
        const suggestionElement = screen.getByText(/Banana/i);
        fireEvent.click(suggestionElement);
        expect(inputElement).toHaveValue('Banana');
      }, 500);
});

test('clears suggestions after selecting an item', () => {
  render(<App />);
  const inputElement = screen.getByPlaceholderText('Type to search...');
  fireEvent.change(inputElement, { target: { value: 'C' } });
  setTimeout(() => {
      const suggestionElement = screen.getByText(/Cherry/i);
      fireEvent.click(suggestionElement);
      const suggestionsList = screen.queryByText(/Cherry/i);
      expect(suggestionsList).toBeNull();
    }, 500);
});

test('matches the snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});