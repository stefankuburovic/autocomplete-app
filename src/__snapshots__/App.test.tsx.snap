// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      AutoComplete Example
    </h1>
    <div
      class="autocomplete"
    >
      <input
        placeholder="Type to search..."
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
